### PR TITLE
fix: Input.Search ConfigProvider size

### LINKED ISSUE #23328: <ConfigProvider componentSize={"large"}> not working on <Input.Search />
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://codepen.io/watsize/pen/pojgXyP](https://codepen.io/watsize/pen/pojgXyP)

### Steps to reproduce
1. View Example in https://codepen.io/watsize/pen/pojgXyP
2. See Input.Search, Search, and Button Components for validating this issue (componentSize will not work on Input.Search)

### What is expected?
Input.Search can be setting by ConfigProvider

### What is actually happening?
Input.Seach is ignore componentSize={"large"} property of ConfigProvider

| Environment | Info |
|---|---|
| antd | 4.1.2 |
| React | 16.13.1 |
| System | Mac OS X 10.15.4 |
| Browser | Google Chrome 80.0.3987.163 |

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #23328

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix: Input.Search ConfigProvider size        |
| 🇨🇳 Chinese | 修复`Input.Search`全局设置size不生效问题          |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/Search.tsx [hunk 1]
```diff
@@ -177,32 +178,39 @@ export default class Search extends React.Component<SearchProps, any> {
...
-        {...restProps}
-        size={size}
-        prefixCls={inputPrefixCls}
-        addonAfter={this.renderAddonAfter(prefixCls)}
-        suffix={this.renderSuffix(prefixCls)}
-        onChange={this.onChange}
-        ref={this.saveInput}
-        className={inputClassName}
-      />
+      <SizeContext.Consumer>
```
Comment (yoyo837) [inline components/input/Search.tsx:199]: renderAddonAfter、renderLoading里的size也要处理

**Comment by afc163** [comment]
来个用例吧。

File: components/config-provider/__tests__/components.test.js [hunk 2]
```diff
@@ -82,6 +82,26 @@ describe('ConfigProvider', () => {
...
+        });
+
+        it('configProvider componentSize middle', () => {
+          expect(
+            render(
+              <ConfigProvider componentSize="middle" prefixCls="config">
+                {renderComponent({})}
+              </ConfigProvider>,
+            ),
+          ).toMatchSnapshot();
```
Comment (yoyo837) [inline components/config-provider/__tests__/components.test.js:103]: 判断一下class 加载对不对

File: components/config-provider/__tests__/components.test.js [hunk 3]
```diff
@@ -82,6 +82,26 @@ describe('ConfigProvider', () => {
...
         });
+
+        it('configProvider componentSize large', () => {
+          expect(
+            render(
+              <ConfigProvider componentSize="large" prefixCls="config">
+                {renderComponent({})}
+              </ConfigProvider>,
+            ),
+          ).toMatchSnapshot();
```
Comment (yoyo837) [inline components/config-provider/__tests__/components.test.js:93]: 判断一下class 加载对不对

Comment (yoyo837) [reply components/config-provider/__tests__/components.test.js:93]: 注意所有的组件都使用这个抽取，不止一个组件

Comment (afc163) [reply components/config-provider/__tests__/components.test.js:93]: 对应的 Snapshot 文件在哪里

Comment (hengkx) [reply components/config-provider/__tests__/components.test.js:93]: components/config-provider/__tests__/__snapshots__/components.test.js.snap
文件是在这个里面
@yoyo837 生成snapshots了 不就是可以判断了吗？
因为之前单元测试没有覆盖到全局默认配置size 所以我给所有组件全局配置都加了size的`snapshots`

**Comment by zombieJ** [comment]
https://ant.design/components/config-provider-cn/#components-config-provider-demo-size 可以加个 Input.Search

**Pull request merged**